core/posts.go

```diff
@@ -43,18 +56,18 @@ type postImage struct {
 	Medium string `json:"medium"`
 }
 
-//GeneratePostSlug  [Create a slug for the post based on the title, if a slug is missing]
-func (n *OpenBazaarNode) GeneratePostSlug(title string) (string, error) {
-	title = strings.Replace(title, "/", "", -1)
-	slugFromTitle := func(title string) string {
+//GeneratePostSlug  [Create a slug for the post based on the status, if a slug is missing]
+func (n *OpenBazaarNode) GeneratePostSlug(status string) (string, error) {
+	status = strings.Replace(status, "/", "", -1)
+	slugFromStatus := func(status string) string {
 		l := SentenceMaxCharacters - SlugBuffer
-		if len(title) < SentenceMaxCharacters-SlugBuffer {
-			l = len(title)
+		if len(status) < SentenceMaxCharacters-SlugBuffer {
+			l = len(status)
 		}
-		return url.QueryEscape(sanitize.Path(strings.ToLower(title[:l])))
+		return url.QueryEscape(sanitize.Path(strings.ToLower(status[:l])))
 	}
```

Comment:
We use the same function when generating a `slug` for a listing (`core/listings.go:115`).

Reply:
Then we should extract it and use the extracted function in all the places it's needed instead of duplicating our logic. (Rationale: https://www.drycode.io/)

Reply:
Can you provide an example of how to do that?